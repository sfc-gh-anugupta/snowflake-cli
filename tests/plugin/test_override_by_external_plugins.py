import pytest


def test_override_build_in_commands(runner, test_root_path, _install_plugin, snapshot):
    config_path = (
        test_root_path / "test_data" / "configs" / "override_plugin_config.toml"
    )

    result = runner.invoke(["--config-file", config_path, "connection", "list"])

    assert result.output == snapshot


def test_disabled_plugin_is_not_executed(
    runner, test_root_path, _install_plugin, snapshot
):
    config_path = (
        test_root_path
        / "test_data"
        / "configs"
        / "disabled_override_plugin_config.toml"
    )

    result = runner.invoke(["--config-file", config_path, "connection", "list"])

    assert result.output == snapshot


@pytest.fixture(scope="module")
def _install_plugin(test_root_path):
    import subprocess

    path = test_root_path / ".." / "test_external_plugins" / "override_build_in_command"
    subprocess.check_call(["pip", "install", path])
