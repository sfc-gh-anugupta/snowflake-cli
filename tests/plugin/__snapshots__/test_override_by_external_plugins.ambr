# serializer version: 1
# name: test_disabled_plugin_is_not_executed
  '''
  +----------------------------------------------------+
  | connection_name | parameters          | is_default |
  |-----------------+---------------------+------------|
  | test            | {'account': 'test'} | False      |
  +----------------------------------------------------+
  
  '''
# ---
# name: test_override_build_in_commands
  '''
  Outside command code
  +----------------------------------------------------+
  | connection_name | parameters          | is_default |
  |-----------------+---------------------+------------|
  | test            | {'account': 'test'} | False      |
  +----------------------------------------------------+
  
  '''
# ---
